tests/test_unreachable.py

```diff
@@ -0,0 +1,65 @@
+from . import v
+
+
+def check_unreachable_lines(assertion, asserted):
+    unreachable_lines = [node.lineno for node in assertion]
+    assert unreachable_lines == asserted
+
+
+def test_assignment(v):
+    v.scan("""\
+def foo():
+    print("Hello World")
+    return
+    a = 1
+""")
+    check_unreachable_lines(v.unreachable_code, [4])
+
+
+def test_multiline_return_statements(v):
+    v.scan("""\
+def foo():
+    print("Something")
+    return (Something,
+            that,
+            spans,
+            over,
+            multiple,
+            lines)
+    print("Hello World")
+""")
+    check_unreachable_lines(v.unreachable_code, [9])
+
+
+def test_multiple_return_statements(v):
+    v.scan("""\
+def foo():
+    return something
+    return None
+    return (Some,
+           big,
+           statemnt)
+""")
+    check_unreachable_lines(v.unreachable_code, [3, 4])
+
+
+def test_pass(v):
+    v.scan("""\
+def foo():
+    return
+    pass
+    return Something
+""")
+    check_unreachable_lines(v.unreachable_code, [3])
+
+
+def test_recursive_functions(v):
+    v.scan("""\
+def foo(a):
+    if a == 1:
+        return 1
+    else:
+        return foo(a-1)
+        print("This line is never executed")
+""")
+    check_unreachable_lines(v.unreachable_code, [6])
```

Comment:
I think we're already on the same page, but just to be clear: we only want to catch unreachable code after return statements in this PR, we'll cover the other cases later.

For tests, we need to think about edge cases and stuff that can go wrong with the current code. Multiple return statements is a good idea for a test case and it uncovered a design error already (see my `return` comment on the code). What other edge cases are there?

```
def foo():
    return; a = 1
```
```
def foo():
    return
    a[1:2]
```

```
class Foo():
    return
    a = 1
```

I don't think we need the additional tests you mention in your message.